internal/pkg/deploy/cloudformation/stack/app.go

```diff
@@ -33,17 +33,51 @@ type AppResourcesConfig struct {
 	Version  int                   `yaml:"Version"`
 }
 
-// AppResourcesService is a configuration for service for a deployed Application StackSet
+// AppResourcesService is a union type of either legacy or new
+// service configuration for a deployed Application StackSet
 type AppResourcesService struct {
+	Name   *LegacyAppResourcesServiceConfig `yaml:"Name,flow"`
```

Comment:
What about this:
```suggestion
	NameFromLegacyConfig   *string  // Deprecated
```
I am not sure if we need the YAML tag, since the legacy metadata would not have `Name` in it.

In fact, I'm not even sure if the legacy config needs to be in `AppResourcesService`. Could we just:

```go
func (s *AppResourcesService) UnmarshalYAML(value *yaml.Node) error {
	if err := value.Decode(&s.Config); err != nil {
		switch err.(type) {
		case *yaml.TypeError:
			break
		default:
			return err
		}
	}
	if !s.Config.isEmpty() {
		return nil
	}
        var legacyName string
	if err := value.Decode(&legacyName); err != nil {
		return err
	}
	// Switch to new app resource service configuration and unset legacy configuration.
	s.Config = AppResourcesServiceConfig{
		Name:    legacyName,
		WithECR: true,
	}
	return nil
}
```

Reply:
it seems like, other than `Unmarshal()`, we _never_ want to use `Name` - they should _always_ use `Config.Name`. maybe unexport it, since it's not needed to unmarshal+marshal anyways?